src/nix_expr.ml

```diff
@@ -89,12 +109,30 @@ let write dest (t:t) =
 		in
 		let property name = put ("." ^ (escape_key name)) in
 
+		let write_inherit expr keys =
+			nl ();
+			put "inherit";
+			space ();
+			begin match expr with
+				| None -> ();
+				| Some expr ->
+					put "(";
+					_write expr;
+					put ")";
+					space ();
+			end;
+			pp_print_list pp_print_string ~pp_sep:pp_print_space formatter keys;
```

Comment:
I don't see anything preventing us from creating an empty list, and I don't think that'll output valid nix. Omit the whole expression if `keys` is empty?

Reply:
done